server-worker/src/main/java/com/aliyun/emr/rss/service/deploy/worker/storage/PartitionFilesSorter.java

```diff
@@ -415,24 +414,30 @@ static class ShuffleBlockInfo {
 
   class FileSorter {
     private final File originFile;
-    private final String sortedFileName;
-    private final String indexFileName;
+    private final File sortedFile;
+    private final File indexFile;
     private final long originFileLen;
     private final String fileId;
     private final String shuffleKey;
 
     FileSorter(File originFile, long originFileLen, String fileId, String shuffleKey) {
       this.originFile = originFile;
-      this.sortedFileName = originFile.getAbsolutePath() + SORTED_SUFFIX;
-      this.indexFileName = originFile.getAbsolutePath() + INDEX_SUFFIX;
+      this.sortedFile = new File(originFile.getParentFile(), originFile.getName() + SORTED_SUFFIX);
+      this.indexFile = new File(originFile.getParentFile(), originFile.getName() + INDEX_SUFFIX);
       this.originFileLen = originFileLen;
       this.fileId = fileId;
       this.shuffleKey = shuffleKey;
+      if (sortedFile.exists()) {
```

Comment:
The last step of sorting a file is to delete the origin file. So I think the condition can be changed to check whether the original file exists or not. If the origin file does not exist, that means sort probably succeeds.

Reply:
> The last step of sorting a file is to delete the origin file. So I think the condition can be changed to check whether the original file exists or not. If the origin file does not exist, that means sort probably succeeds.

For this pr, I am considering the case that originFile and sortedFiles both exist, and the `sortedFiles` map recovered from LevelDB not contain this fils's information. Then new `openStream` request come,  then we will write to a existed sortedFile.